Intermittent failures in MQTT Unit Tests

![Screenshot from 2021-09-21 09-04-52](https://user-images.githubusercontent.com/1342980/134185706-e40bda0c-4536-4c8c-8c6e-2b4e4fce0076.png)

This particular screen grab shows all three tests failing, but sometimes it's one or two. Sometimes all of them pass. I think this is b/c of the config file being used which instructs the provider to connect to an actual MQTT endpoint (see test/java/com/alvarium/streams/mqtt-config.json).

Consider removing these tests as they aren't unit tests. The MQTT Provider requires an actual broker so these are more like integration tests. We won't be guaranteed to have an MQTT endpoint in our build pipeline, so these tests can fail which will fail the build pipeline.

Output from `mqttShouldConnect` test
```
com.alvarium.streams.StreamException: Connection failed

	at com.alvarium.streams.MqttStreamProvider.reconnect(MqttStreamProvider.java:77)
	at com.alvarium.streams.MqttStreamProvider.connect(MqttStreamProvider.java:40)
	at com.alvarium.streams.MqttStreamProviderTest.mqttShouldConnect(MqttStreamProviderTest.java:34)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.junit.runners.Suite.runChild(Suite.java:127)
	at org.junit.runners.Suite.runChild(Suite.java:26)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: Connection lost (32109) - java.io.EOFException
	at org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:197)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.EOFException
	at java.base/java.io.DataInputStream.readByte(DataInputStream.java:272)
	at org.eclipse.paho.client.mqttv3.internal.wire.MqttInputStream.readMqttWireMessage(MqttInputStream.java:92)
	at org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:137)
	... 1 more
```

--- a/src/test/java/com/alvarium/streams/MqttStreamProviderTest.java
+++ b/src/test/java/com/alvarium/streams/MqttStreamProviderTest.java
@@ -1,5 +1,8 @@
 package com.alvarium.streams;
 
+import static org.junit.Assert.assertNotNull;
+
+import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Paths;
@@ -9,17 +12,34 @@ import com.alvarium.SdkAction;
 
 import org.junit.Test;
 
+/**
+ * Tests for connecting with the MQTT broker are not annotated with `@Test` so these tests do not 
+ * run during unit testing. This is because no MQTT endpoint is guranteed in the build pipeline for
+ * the SDK.
+ * 
+ * This class does include tests for loading MQTT configuration
+ */
 public class MqttStreamProviderTest {
-  final StreamInfo info;
   
-  public MqttStreamProviderTest() throws Exception {
-    String path = "./src/test/java/com/alvarium/streams/mqtt-config.json";
-    String testJson = Files.readString(Paths.get(path), StandardCharsets.US_ASCII);
-    info = StreamInfo.fromJson(testJson);
+  // Test that the config-loaded configuration can actually be casted to MqttConfig type
+  @Test
+  public void mqttShouldLoadConfig() throws StreamException {
+    final String path = "./src/test/java/com/alvarium/streams/mqtt-config.json";
+    final String testJson;
+    try {
+      testJson  = Files.readString(Paths.get(path), StandardCharsets.US_ASCII);
+    } catch(IOException e) {
+      throw new StreamException("Could not read configuration file", e);
+    }
+    final StreamInfo info = StreamInfo.fromJson(testJson);
+    final MqttConfig config = MqttConfig.class.cast(info.getConfig());
+    assertNotNull(config);
   }
 
-  @Test
-  public void mqttShouldPublish() throws StreamException {
+  public void mqttShouldPublish() throws Exception {
+    String path = "./src/test/java/com/alvarium/streams/mqtt-config.json";
+    String testJson = Files.readString(Paths.get(path), StandardCharsets.US_ASCII);
+    StreamInfo info = StreamInfo.fromJson(testJson);
     StreamProviderFactory factory = new StreamProviderFactory();
     StreamProvider mqttProvider =  factory.getProvider(info);
     String msg = "hello";
@@ -27,15 +47,19 @@ public class MqttStreamProviderTest {
     mqttProvider.publish(wrapper);
   }
 
-  @Test
-  public void mqttShouldConnect() throws StreamException {
+  public void mqttShouldConnect() throws Exception {
+    String path = "./src/test/java/com/alvarium/streams/mqtt-config.json";
+    String testJson = Files.readString(Paths.get(path), StandardCharsets.US_ASCII);
+    StreamInfo info = StreamInfo.fromJson(testJson);
     StreamProviderFactory factory = new StreamProviderFactory();
     StreamProvider mqttProvider =  factory.getProvider(info);
     mqttProvider.connect();
   }
 
-  @Test
   public void mqttShouldClose() throws Exception {
+    String path = "./src/test/java/com/alvarium/streams/mqtt-config.json";
+    String testJson = Files.readString(Paths.get(path), StandardCharsets.US_ASCII);
+    StreamInfo info = StreamInfo.fromJson(testJson);
     StreamProviderFactory factory = new StreamProviderFactory();
     StreamProvider mqttProvider =  factory.getProvider(info);
     mqttProvider.connect();

